app/views/npq_separation/admin/applications/history/show.html.erb

```diff
@@ -25,4 +25,11 @@
 
 <br>
 
-<%= render NpqSeparation::Admin::HistoryComponent.new(record: @application) %>
+<%=
+  render(NpqSeparation::Admin::HistoryComponent.new(record: @application) do |application, version_created_at, object_changes|
+    if object_changes["training_status"].present?
+      reason = ApplicationState.lookup_reason(application:, created_at: version_created_at, state: object_changes["training_status"].last)
+    end
+    "Reason for training status change: #{reason}" if reason
```

Comment:
This is doing multiple database lookups?  if you move the logic to Application go via the `application_states` relationship you'll avoid the repeated lookups?

```ruby
do |application, version_created_at, object_changes|
    if object_changes["training_status"].present?
      reason = @application.lookup_state_change_reason(created_at: version_created_at, state: object_changes["training_status"].last)
    end
end

class Application
  def lookup_state_change_reason
    application_states.find do |application_state|
      application_state.created_at > changed_at - variance &&
        application_state.created_at < changed_at + variance &&
        application_state.training_status == changed_status
    end
  end
end
```